Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider

```diff
@@ -166,3 +166,65 @@ AS
 
 	EXEC {databaseOwner}{objectQualifier}EnsureLocalizationExists @PortalID = @PortalId, @CultureCode = @CultureCode
 GO
+
+-- DNN-8358: Improve Journal_Save to support updates
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}Journal_Save]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}Journal_Save]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}Journal_Save]
+    @PortalID           Int,
+    @JournalId          Int,
+    @JournalTypeId      Int,
+    @UserID             Int,
+    @ProfileId          Int,
+    @GroupId            Int,
+    @Title              nVarChar(255),
+    @Summary            nVarChar(2000),
+    @ItemData           nVarChar(2000),
+    @JournalXML         xml,
+    @ObjectKey          nVarChar(255),
+    @AccessKey          UniqueIdentifier,
+    @SecuritySet        nVarChar(2000),
+    @CommentsDisabled   Bit,
+    @CommentsHidden     Bit
+AS
+BEGIN
+    IF IsNull(@JournalId, -1) = -1 BEGIN
```

Comment:
New to the DNN way of doing things... but can't we wrap this in a `BEGIN TRAN` / `COMMIT TRAN` so we don't end up modifying the `Journal`, then not finishing the `JournalSecurity` rows?